lib/ansible/plugins/connection/psrp.py

```diff
@@ -432,50 +443,20 @@ def exec_command(self, cmd: str, in_data: bytes | None = None, sudoable: bool =
         super(Connection, self).exec_command(cmd, in_data=in_data,
                                              sudoable=sudoable)
 
-        pwsh_in_data: bytes | str | None = None
+        pwsh_in_data = to_text(in_data, errors="surrogate_or_strict", nonstring="passthru")
```

Comment:
If you drop the explicit type annotation, MyPy will infer it from the expression return value being assigned. `to_text(...) -> str` means that at this point, MyPy will memorize that `pwsh_in_data` is of type `str`. And so it fails later on line 460 that tries to assign `in_data: bytes | None` to this variable which is incompatible with `str`.

You have to either keep the annotation allowing all possible values as follows
```suggestion
        pwsh_in_data: bytes | str | None = to_text(in_data, errors="surrogate_or_strict", nonstring="passthru")
```
or normalize the value assigned on line 460 to `str` in order to make MyPy pass.